apps/src/templates/progress/googlePlatformApiRedux.js

```diff
@@ -0,0 +1,139 @@
+import firehoseClient from '@cdo/apps/lib/util/firehose';
+
+/*
+ * The Google Classroom Share Button is only available through the google
+ * platform api, so we have to add it to our page on load. We manage loading
+ * in redux since multiple components need to observe the loading state.
+ */
+
+const START_LOADING_GAPI = 'googlePlatformApi/START_LOADING_GAPI';
+const FINISH_LOADING_GAPI = 'googlePlatformApi/FINISH_LOADING_GAPI';
+
+const startLoadingGapi = startTime => ({type: START_LOADING_GAPI, startTime});
+const finishLoadingGapi = success => ({type: FINISH_LOADING_GAPI, success});
+
+export const GOOGLE_PLATFORM_API_ID = 'GooglePlatformApiId';
+const LOAD_TIMEOUT_MILLIS = 15000;
+
+const initialState = {
+  loading: false,
+  loaded: false,
+  loadStartTime: null
```

Comment:
loadStartTime looks like it's acting as a local variable. It shouldn't be in the global redux store.

Reply:
can you explain why it looks like it's acting as a local variable? it's bound to the state, which is keeping track of the state of the api loading process, including when it starts.

Reply:
Looks like that is true for `loading` as well

Reply:
(oops missed your comment until just now)
By "local variable" I meant that I don't think I see it being used anywhere outside `googlePlatformApiRedux` yet since it's in the redux store, I could be accessed from anywhere in the application.

Reply:
i'm confused about why that's an issue. redux doesn't support the concept of internal properties so these are implicitly exposed in the redux tree, but that doesn't change the fact that they _are_ necessary properties of the state of the application that the loading process needs to be able to query.

Reply:
Same as my above redux comment (changing to non-blocking)